tests/cupy_tests/core_tests/test_ndarray_scatter.py

```diff
@@ -0,0 +1,95 @@
+import unittest
+
+import numpy
+
+import cupy
+from cupy import testing
+
+
+@testing.parameterize(
+    # array only
+    {'shape': (2, 3, 4), 'slices': numpy.array(-1), 'value': 1},
+    {'shape': (2, 3, 4), 'slices': numpy.array([1, 0]), 'value': 1},
+    {'shape': (2, 3, 4), 'slices': (slice(None), [1, 2]), 'value': 1},
+    {'shape': (3, 4, 5),
+     'slices': (slice(None), [[1, 2], [0, -1]],), 'value': 1},
+    {'shape': (3, 4, 5),
+     'slices': (slice(None), slice(None), [[1, 2], [0, 3]]), 'value': 1},
+    # slice and array
+    {'shape': (3, 4, 5),
+     'slices': (slice(None), slice(1, 2), [[1, 3], [0, 2]]), 'value': 1},
+    # None and array
+    {'shape': (3, 4, 5),
+     'slices': (None, [1, -1]), 'value': 1},
+    {'shape': (3, 4, 5),
+     'slices': (None, [1, -1], None), 'value': 1},
+    {'shape': (3, 4, 5),
+     'slices': (None, None, None, [1, -1]), 'value': 1},
+    # None, slice and array
+    {'shape': (3, 4, 5),
+     'slices': (slice(0, 1), None, [1, -1]), 'value': 1},
+    {'shape': (3, 4, 5),
+     'slices': (slice(0, 1), slice(1, 2), [1, -1]), 'value': 1},
+    {'shape': (3, 4, 5),
+     'slices': (slice(0, 1), None, slice(1, 2), [1, -1]), 'value': 1},
+    # broadcasting
+    {'shape': (3, 4, 5), 'slices': (slice(None), [[1, 2], [0, -1]],),
+     'value': numpy.arange(3 * 2 * 2 * 5).reshape(3, 2, 2, 5)},
+)
```

Comment:
There is a test case that handles `cupy.core.core.ndarray` indices. It is `test_scatter_add_cupy_argument` locating at line 75 of `test_ndarray_scatter.py`.